adapter/internal/operator/synchronizer/synchronizer.go

```diff
@@ -121,23 +132,24 @@ func deleteAPIFromEnv(httpRoute *gwapiv1b1.HTTPRoute, apiState APIState) error {
 func deployAPIInGateway(apiState APIState) error {
 	var err error
 	if apiState.ProdHTTPRoute != nil {
-		_, err = GenerateMGWSwagger(apiState, apiState.ProdHTTPRoute, true)
+		_, err = GenerateMGWSwagger(apiState, apiState.ProdHTTPRoute, true, getHostNameForBackend)
```

Comment:
why do you pass a function through multiple methods as a method param? why can't we resolve the backend url at resolveref step in APIcontrollers reconcile? then you don't have to add client etc too to synchronizer

Reply:
Since we are planning to remove APIState I avoid adding more state data to APIState. can add a 2d string array to map resolved hostnames there and go with that as well

Reply:
Updated the changes to set the backend properties to apiState